src/imp/openssl.rs

```diff
@@ -266,6 +266,16 @@ impl TlsConnector {
             connector.cert_store_mut().add_cert((cert.0).0.clone())?;
         }
 
+        let alpn = builder
+            .alpn_protocols
+            .iter()
+            .map(|proto| format!("\\x{:x}{}", proto.len(), proto))
```

Comment:
The sequence that openssl wants is literally the byte length of the protocol followed by it, not the string `\xN`.

Reply:
Ah, my eyes tricked me when reading https://github.com/sfackler/hyper-openssl/blob/master/src/lib.rs#L107.